float32 speed_B 
float32 speed_C
